### DISCUSSION TITLE
Could webots using autosave?

### DISCUSSION THREAD

**Discussion by Lr-2002** [cyberbotics/webots#3958] [Ideas]
Background: When modeling in webots without saving ,I run the fiel. But the window closed imediatly without save and my file gone.

So could you change the save mechanism or add the auto save ?

**Comment by omichel** [comment]
Maybe we could display a warning if a user attempts to run a simulation after she made some modification in the scene tree and didn't save the world. The warning dialog could display "The world file was modified but not saved, are you sure you want to run the simulation? Saving it after running the simulation is not recommended.". The dialog should include buttons to "save", "proceed without saving" or "cancel". Would that resolve the problem?

**Reply by Lr-2002** [reply]
Absolutely you are right.
But i'm interested in that why is there no autosave mode in WEBOTS？
Maybe it cause some problem？
If not,i'd like to code fot it because i take its existence for granted.😭
(if my English bothered you,sorry for being not a native speaker.)

**Reply by omichel** [reply]
Depending on the size of the world file, auto-save may be time consuming and could make Webots non-responsive while it is saving the world file. So, if we introduce this option, we should benchmark it on several world files with different sizes and render it optional in the Webots preferences. If the non-responsive behavior is too much annoying, we may decide to drop this feature and revert to my first proposal. Anyhow, if you like, feel free to implement such a feature by yourself. We will be happy to review your contribution in a new PR and then possibly merge it in the develop branch of Webots.

**Reply by Lr-2002** [reply]
WOW,thanks for your suggestion and support！
I'll do it after finishing my nowadays work！

**Reply by taoudi-abdelbasset** [reply]
any progress with autosave??
